app/controllers/thread_controller.php

```diff
@@ -72,5 +131,56 @@ public function create()
         $this->set(get_defined_vars());                    
         $this->render($page);     
     }
+
+    public function write()                        
+    {
+        $thread = Thread::get(Param::get('thread_id'));
+        $comment = new Comment;
+        $page = Param::get('page_next', 'write');
+        switch ($page) {
+        case 'write':                    
+            break;
+        case 'write_end':                
+            $comment->username = Param::get('username');
+            $comment->body = Param::get('body');
+            try {            
```

Comment:
The case statement MUST be indented once from switch, and the break keyword (or other terminating keyword) MUST be indented at the same level as the case body.

```switch ($page) {
    case 'write':
        break;
    case 'write_end':
        echo 'Second case, which falls through';
        // no break
    case 2:
       $comment->username = Param::get('username');
       $comment->body = Param::get('body');
       ....
    default:
        ...
        break;
}```